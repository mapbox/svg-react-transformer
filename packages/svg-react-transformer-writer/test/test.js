'use strict';

const tempy = require('tempy');
const fs = require('fs');
const pify = require('pify');
const path = require('path');
const del = require('del');
const svgReactTransformer = require('@mapbox/svg-react-transformer');
const writer = require('..');

describe('svgReactTransformerWriter', () => {
  let tmpDir;
  beforeEach(() => {
    tmpDir = tempy.directory();
  });
  afterEach(() => {
    return del(tmpDir, { force: true });
  });

  test('outputs one component file per SVG source file', () => {
    return writer(path.join(__dirname, './fixtures/**'), tmpDir)
      .then(() => pify(fs.readdir)(tmpDir))
      .then(outputFiles => {
        expect(outputFiles).toEqual(['airplane.js', 'apple.js', 'big.js']);
      });
  });

  test('file content matches svgReactTransformer.toComponentModule, with name provided', () => {
    const fixtureContent = fs.readFileSync(
      path.join(__dirname, 'fixtures/apple.svg'),
      'utf8'
    );
    return writer([path.join(__dirname, 'fixtures/apple.svg')], tmpDir)
      .then(() => {
        return Promise.all([
          pify(fs.readFile)(path.join(tmpDir, 'apple.js'), 'utf8'),
          svgReactTransformer.toComponentModule(fixtureContent, {
            name: 'apple'
          })
        ]);
      })
      .then(data => {
        expect(data[0]).toBe(data[1]);
      });
  });

  test('options.filenameTemplate', () => {
    const options = {
      filenameTemplate: name => `svg-cmp-${name}`
    };
    return writer(path.join(__dirname, './fixtures/**'), tmpDir, options)
      .then(() => pify(fs.readdir)(tmpDir))
      .then(outputFiles => {
        expect(outputFiles).toEqual([
          'svg-cmp-airplane.js',
          'svg-cmp-apple.js',
          'svg-cmp-big.js'
        ]);
      });
  });

  test('options passed to toComponentModule', () => {
    const options = {
      filenameTemplate: name => `svg-cmp-${name}`,
      svgoPlugins: [{ removeDimensions: true }],
      propTypes: `{ name: 'PropTypes.string.isRequired' }`,
      defaultProps: `{ name: 'foo' }`
    };

    return writer(path.join(__dirname, './fixtures/apple.svg'), tmpDir, options)
      .then(() =>
        pify(fs.readFile)(path.join(tmpDir, 'svg-cmp-apple.js'), 'utf8')
      )
      .then(result => {
        expect(result).toMatchSnapshot();
      });
  });
});
