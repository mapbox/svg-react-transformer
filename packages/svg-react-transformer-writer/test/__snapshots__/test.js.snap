// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svgReactTransformerWriter options passed to toComponentModule 1`] = `
"
    'use strict';
    const React = require('react');
    const PropTypes = require('prop-types');

    class Apple extends React.PureComponent {
      render() { return <svg viewBox=\\"0 0 18 18\\" {...this.props}><path d=\\"M3.4 9.2c.1-1.1.4-2.3 1.5-3.2.9-.7 1.8-.8 2.9-.5 1.4.5 1 .5 2.4 0 1.5-.6 3 0 3.7.8.1.2.2.3 0 .4-1.8 1.3-1.6 3.9.3 5 .2.1.3.2.2.4-.4 1-1 1.9-1.7 2.7-.5.5-1.1.7-1.8.4-.1 0-.3-.1-.4-.1-.9-.1-1.7-.1-2.5.2-.3.1-.5.2-.8.3-.4.1-.8-.1-1.1-.4-.7-.5-1.1-1.2-1.5-1.9-.7-1.1-1.2-2.7-1.2-4.1\\"/><path d=\\"M11.6 2.5c0 1.2-1 2.4-2 2.7-.6.1-.8 0-.7-.6.2-1.2 1.2-2.2 2.5-2.5.2 0 .2 0 .2.2v.2\\"/></svg>; }
    }
    
Apple.propTypes = { name: 'PropTypes.string.isRequired' }

Apple.defaultProps = { name: 'foo' }

    module.exports = Apple;
  "
`;
